kani-compiler/src/codegen_aeneas_llbc/mir_to_ullbc/mod.rs

```diff
@@ -152,31 +182,184 @@ impl<'a, 'tcx> Context<'a, 'tcx> {
         tid.try_into().unwrap()
     }
 
+    fn register_type_decl_id(&mut self, def_id: DefId) -> CharonTypeDeclId {
+        debug!("register_type_decl_id: {:?}", def_id);
+        let tid = match self.id_map.get(&def_id) {
+            Some(tid) => *tid,
+            None => {
+                debug!("***Not found!");
+                let tid = CharonAnyTransId::Type(self.translated.type_decls.reserve_slot());
+                self.id_map.insert(def_id, tid);
+                self.translated.all_ids.insert(tid);
+                tid
+            }
+        };
+        debug!("register_type_decl_id: {:?}", self.id_map);
+        tid.try_into().unwrap()
+    }
+
+    // similar to register_type_decl_id, but not adding new def_id, used for cases where the def_id has been registered, or in functions that take immut &self
+    fn find_type_decl_id(&self, def_id: DefId) -> CharonTypeDeclId {
```

Comment:
As discussed offline, please rename this to `get_type_decl_id` as `find` sounds infallible, but this function panics if the ID is not found.